daycount_test.go

```diff
@@ -14,558 +15,215 @@ const epsilon = 1.0e-6
-	t.Parallel()
-
-	for _, tc := range []struct {
-		name     string
-		from     date.Date
-		to       date.Date
-		expected float64
-	}{
-		{
-			"from 2007.12.28 to 2008.02.28",
-			date.New(2007, time.December, 28),
-			date.New(2008, time.February, 28),
-			62.0 / 360.0,
-		},
-		{
-			"from 2007.12.28 to 2008.02.29",
-			date.New(2007, time.December, 28),
-			date.New(2008, time.February, 29),
-			63.0 / 360.0,
-		},
-		{
-			"from 2007.10.31 to 2008.11.30",
-			date.New(2007, time.October, 31),
-			date.New(2008, time.November, 30),
-			396.0 / 360.0,
-		},
-		{
-			"from 2007.02.01 to 2008.05.31",
-			date.New(2007, time.February, 1),
-			date.New(2008, time.May, 31),
-			485.0 / 360.0,
+			map[Convention]float64{
+				ActualActual: 100.0,
+			},
 		},
 	} {
 		tc := tc
-		t.Run(tc.name, func(t *testing.T) {
-			t.Parallel()
 
-			assert.InDelta(t, tc.expected, yearFractionActualThreeSixty(tc.from, tc.to), epsilon)
-		})
-	}
-}
+		for convention, expected := range tc.expected {
+			convention, expected := convention, expected
 
-func Test_yearFractionActualThreeSixtyFiveFixed(t *testing.T) {
-	t.Parallel()
+			name := fmt.Sprintf("%s to %s/%s", tc.from, tc.to, convention)
+			t.Run(name, func(t *testing.T) {
+				t.Parallel()
 
-	for _, tc := range []struct {
-		name     string
-		from     date.Date
-		to       date.Date
-		expected float64
-	}{
-		{
-			"from 2007.12.28 to 2008.02.28",
-			date.New(2007, time.December, 28),
-			date.New(2008, time.February, 28),
-			62.0 / 365.0,
-		},
-		{
-			"from 2007.12.28 to 2008.02.29",
-			date.New(2007, time.December, 28),
-			date.New(2008, time.February, 29),
-			63.0 / 365.0,
-		},
-		{
-			"from 2007.10.31 to 2008.11.30",
-			date.New(2007, time.October, 31),
-			date.New(2008, time.November, 30),
-			396.0 / 365.0,
-		},
-		{
-			"from 2007.02.01 to 2008.05.31",
-			date.New(2007, time.February, 1),
-			date.New(2008, time.May, 31),
-			485.0 / 365.0,
-		},
-	} {
-		tc := tc
-		t.Run(tc.name, func(t *testing.T) {
-			t.Parallel()
-
-			assert.InDelta(t, tc.expected, yearFractionActualThreeSixtyFiveFixed(tc.from, tc.to), epsilon)
-		})
+				assert.InEpsilon(t, expected, YearFraction(tc.from, tc.to, convention), epsilon)
+			})
+		}
 	}
 }
 
-func Test_yearFractionThirtyThreeSixtyUS(t *testing.T) {
+func Test_YearFraction_EqualDates(t *testing.T) {
 	t.Parallel()
 
-	for _, tc := range []struct {
-		name     string
-		from     date.Date
-		to       date.Date
-		expected float64
-	}{
-		{
-			"from 2007.12.28 to 2008.02.28",
-			date.New(2007, time.December, 28),
-			date.New(2008, time.February, 28),
-			60.0 / 360.0,
-		},
-		{
-			"from 2007.12.28 to 2008.02.29",
-			date.New(2007, time.December, 28),
-			date.New(2008, time.February, 29),
-			61.0 / 360.0,
-		},
-		{
-			"from 2007.10.31 to 2008.11.30",
-			date.New(2007, time.October, 31),
-			date.New(2008, time.November, 30),
-			390.0 / 360.0,
+	var daysFromOrigin int
+	fuzz.NewWithSeed(123).Funcs(
```

Comment:
Since go 1.18, [native go supports fuzzing](https://go.dev/security/fuzz/#glos-fuzz-test) (`func FuzzYearFractionEqualDates(f *testing.F)`). Out of curiosity, if I had to introduce fuzz testing in a repo, should I use native go fuzzing instead of the package gofuzz?

If native go should be preferred, then this PR might or might not be the right place for making this transition.

Reply:
Feel free to give it a go indeed, good point :+1: And yes, I would do this separately in any case